nipype/interfaces/mrtrix3/preprocess.py

```diff
@@ -28,16 +28,17 @@ class DWIDenoiseInputSpec(MRTrix3BaseInputSpec):
         desc='set the window size of the denoising filter. (default = 5,5,5)')
     noise = File(
```

Comment:
it doesn't look like this is currently being tracked as an output by the interface, but if this is set it I think `dwidenoise` would make it. You should :
1. re-add it to the OutputSpec (`noise = File(..., exists=True`)
2. add a `_list_outputs()` method that checks if `self.inputs.noise` is defined, and if so adds it to outputs.

Here's an example of (2) - though you won't have to generate a filename.